profiles/utils.py

```diff
@@ -12,21 +15,90 @@
 from open_discussions.utils import now_in_utc
 
 
+# Max dimension of either height or width for small and medium images
+IMAGE_SMALL_MAX_DIMENSION = 64
+IMAGE_MEDIUM_MAX_DIMENSION = 128
+
+IMAGE_STANDARD = "image"
+IMAGE_SMALL = "image_small"
+IMAGE_MEDIUM = "image_medium"
+
+MAX_IMAGE_FIELD_LENGTH = 1024
+
+IMAGE_FIELDS = {
+    IMAGE_STANDARD: "",
+    IMAGE_SMALL: IMAGE_SMALL_MAX_DIMENSION,
+    IMAGE_MEDIUM: IMAGE_MEDIUM_MAX_DIMENSION
+}
+
 # This is the Django ImageField max path size
 IMAGE_PATH_MAX_LENGTH = 100
 IMAGE_PATH_PREFIX = 'profile'
+GRAVATAR_IMAGE_URL = "https://www.gravatar.com/avatar/{}.jpg"
+
+DEFAULT_FONTS = [
+    'HelveticaNeue-Light',
+    'Helvetica Neue Light',
+    'Helvetica Neue',
+    'Helvetica',
+    'Arial',
+    'Lucida Grande',
+    'sans-serif',
+]
+
+SVG_TEMPLATE = """
+<svg xmlns="http://www.w3.org/2000/svg" pointer-events="none"
+     width="{size}" height="{size}">
+  <circle cx="{cx}" cy="{cy}" r="{radius}" style="{style}"></circle>
+  <text text-anchor="middle" y="50%" x="50%" dy="0.35em"
+        pointer-events="auto" font-family="{font-family}"
+        font-weight="400" fill="#{color}" style="{text-style}">{text}</text>
+</svg>
+""".strip()
+SVG_TEMPLATE = re.sub(r'(\s+|\n)', ' ', SVG_TEMPLATE)
+
+DEFAULT_PROFILE_IMAGE = urljoin(settings.STATIC_URL, "images/avatar_default.png")
+
+
+def generate_gravatar_image(user, image_field=None):
+    """
+    Query gravatar for an image and return those image properties
 
-default_profile_image = urljoin(settings.STATIC_URL, "images/avatar_default.png")
-
+    Args:
+        user (User): the user to compute gravatar image urls for
 
-def image_uri(user, image_field='image_small'):
+    Returns:
+        str: The URL to the image.
+    """
+    gravatar_hash = hashlib.md5(user.email.lower().encode('utf-8')).hexdigest()
+    gravatar_image_url = GRAVATAR_IMAGE_URL.format(gravatar_hash)
+    max_dimension = IMAGE_FIELDS[image_field]
+    size_param = '&s={}'.format(max_dimension) if max_dimension else ''
+    if user.profile.name:
+        d_param = urljoin(
+            settings.SITE_BASE_URL,
+            '/profile/{}/{}/fff/579cf9.png'.format(
+                user.username,
+                max_dimension
+            )
+        )
+    else:
+        d_param = settings.SITE_BASE_URL + DEFAULT_PROFILE_IMAGE
```

Comment:
Just wondering: could/should `urljoin` be used here instead of concatenating via `+`?